pkg/autodiscovery/providers/file_test.go

```diff
@@ -109,11 +110,17 @@ func TestCollect(t *testing.T) {
 	// metric files don't override default files
 	assert.Equal(t, 2, len(get("qux")))
 
+	// logs files don't override default files
+	assert.Equal(t, 2, len(get("corge")))
+
 	// metric files not collected in root directory
 	assert.Equal(t, 0, len(get("metrics")))
 
+	// logs files collected in root directory
+	assert.Equal(t, 1, len(get("logs-agent_only")))
+
 	// total number of configurations found
-	assert.Equal(t, 11, len(configs))
+	assert.Equal(t, 14, len(configs))
```

Comment:
why is it +3 here? I see you're adding 2 configurations below

Reply:
Because we were skipping `logs-agent_only.yaml` with a hard coded rule.